dns_server/src/main.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use clap::Parser;
+use config::DnsServerConfig;
+use crawler::{storage_impl::DnsServerStorageImpl, Crawler};
+use p2p::net::default_backend::{
+    transport::{NoiseEncryptionAdapter, NoiseTcpTransport, TcpTransportSocket},
+    DefaultNetworkingService,
+};
+use tokio::sync::mpsc;
+
+mod config;
+mod crawler;
+mod dns_server;
+mod error;
+
+fn make_p2p_transport() -> NoiseTcpTransport {
+    let stream_adapter = NoiseEncryptionAdapter::gen_new();
+    let base_transport = TcpTransportSocket::new();
+    NoiseTcpTransport::new(stream_adapter, base_transport)
+}
+
+async fn run(config: Arc<DnsServerConfig>) -> Result<void::Void, error::DnsServerError> {
+    let (command_tx, command_rx) = mpsc::unbounded_channel();
+
+    let chain_config = if config.testnet {
+        Arc::new(common::chain::config::create_testnet())
+    } else {
```

Comment:
And after using network ids, we match each possible string. If none matches, we panic.

Reply:
Should be OK now.